README.md

```diff
@@ -1,2 +1,105 @@
+  In this post-processing Phase the Procedure performs the steps needs to report and mitigate failures occurred during the Processing Phase.  
+  Possible steps in this Phase are:
+
+  - Clear the "processing" state
+  - Record error (Log, Track)
+  - May restore state to initial condition (Recover/Undo)
+  - May notify the user
+  - May retry by dispatching a retry Message
+
+* **Processing Succeeded Phase**  
+  In this post-processing Phase the Procedure performs the steps needs to record the success and the results of the Processing Phase.  
+  Possible steps in this Phase are:
+
+  - Clear the "processing" state
+  - Update the model
+  - May notify the user
+
+* **Broadcast Update Phase**  
+  In this last Phase the Procedure broadcasts a Message on what was updated. This may cause other Procedures to perform additional work.  
+  This Phase has a single step:
+
+  - Dispatch item updated message
+
+## Motivations for the "Procedure Module" pattern
+
+- The Phases of message processing are coupled. Changes to one Phase may affect the following Phase(s).
+- Placing all Phases of a message processing Procedure inside a Module increases the cohesion of the code.
+- Easily add new Procedures by adding new **Procedure Modules**, without the need to modify the existing ones.
+- It scales well in large teams where several people can work on separate **Procedure Modules** with no or minimal merge conflicts.
+
+## Analogy
+
+The **Organize Code by Features** pattern tells us to group all files that implement one feature under one ancestor folder.
+
+The **Procedure Module** pattern tells us to put all Phases of a message processing Procedure inside one code Module.  
+In this case the feature is the Procedure.
+
+## Testing
+
+In most cases, it is recommend to test the Procedure as a whole (black-box testing), instead of testing each Phase independently (white-box testing).
+
+## Recommendations
+
+### Avoid Orchestration
+
+Orchestration occurs when one creates a "master" multi-step Procedure that controls other Procedures.
+Procedures become tightly coupled with the “master” Procedure gaining too much responsibility and complexity. Adding new Procedures often requires changes to existing ones.
+Only use Orchestration with truly sub-procedures where the "master-procedure" would not be complete with out the "sub-procedure" and the "sub-procedure" will not make sense by itself.
+
+### Embrace Choreography
+
+Choreography occurs when one creates Procedures that observe the Updates broadcasted by other Procedures. Procedures are loosely coupled. It is easy to add new Procedures with no or minimal changes to existing ones.
+
+## History
+
+This pattern emerged in 2005-2006 while working on Rich Internet Applications written in [ActionScript](https://en.wikipedia.org/wiki/ActionScript).  
+[Andrea Wyss](https://www.linkedin.com/in/awyss/) and [Trent DiBacco](https://www.linkedin.com/in/trent-dibacco-82840515/) refer to it as the "Usecase" pattern.
```

Comment:
Why give it another name? Has it been referred to as the "Usecase" pattern in some official capacity?

Reply:
That is what we called it for many years. But we never officially published it.
I only documented in internal wiki in previous company I worked at.
Engineer I worked with know it by that name.